libcontainer/nsenter/cloned_binary.c

```diff
@@ -205,52 +206,58 @@ static int make_execfd(int *fdtype)
 	 * sealing and also doesn't require assumptions like /tmp.
 	 */
 	*fdtype = EFD_MEMFD;
-	fd = memfd_create(RUNC_MEMFD_COMMENT, MFD_CLOEXEC | MFD_ALLOW_SEALING);
-	if (fd >= 0)
-		return fd;
-	if (errno != ENOSYS)
-		goto err;
+	if (*fdtype & allowed) {
```

Comment:
Perhaps it could be better write this as

```C
if (allowed & EFD_MEMFD)
```

not for the sake of optimization -- I hope compiler optimizes the dereference out -- but just for readability.